plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/TestElasticsearchQueryBuilder.java

```diff
@@ -0,0 +1,136 @@
+                                .should(new TermQueryBuilder(name.getName(), "bob"))));
+        // all
+        assertQueryBuilder(
+                ImmutableMap.of(age, Domain.all(INTEGER)),
+                new MatchAllQueryBuilder());
+
+        // notNull
+        assertQueryBuilder(
+                ImmutableMap.of(age, Domain.notNull(INTEGER)),
+                new BoolQueryBuilder().filter(new ExistsQueryBuilder(age.getName())));
+
+        // isNull
+        assertQueryBuilder(
+                ImmutableMap.of(age, Domain.onlyNull(INTEGER)),
+                new BoolQueryBuilder().mustNot(new ExistsQueryBuilder(age.getName())));
+
+        // isNullAllowed
+        assertQueryBuilder(
+                ImmutableMap.of(age, Domain.singleValue(INTEGER, 1L, true)),
+                new BoolQueryBuilder().filter(
+                        new BoolQueryBuilder()
+                                .should(new TermQueryBuilder(age.getName(), 1))
+                                .should(new BoolQueryBuilder().mustNot(new ExistsQueryBuilder(age.getName())))));
+    }
+
+    @Test
+    public void testMultiConstraint()
+    {
+        assertQueryBuilder(
+                ImmutableMap.of(
+                        age, Domain.singleValue(INTEGER, 1L),
+                        score, Domain.create(ValueSet.ofRanges(Range.range(DOUBLE, 65.0, false, 80.0, true)), false)),
+                new BoolQueryBuilder()
+                        .filter(new TermQueryBuilder(age.getName(), 1))
+                        .filter(new RangeQueryBuilder(score.getName()).gt(65.0).lte(80.0)));
+
+        assertQueryBuilder(
+                ImmutableMap.of(
+                        length, Domain.create(ValueSet.ofRanges(Range.range(DOUBLE, 160.0, true, 180.0, true)), false),
+                        score, Domain.create(ValueSet.ofRanges(Range.range(DOUBLE, 65.0, false, 80.0, true)), false)),
+                new BoolQueryBuilder()
+                        .filter(new RangeQueryBuilder(length.getName()).gte(160.0).lte(180.0))
+                        .filter(new RangeQueryBuilder(score.getName()).gt(65.0).lte(80.0)));
+
+        assertQueryBuilder(
+                ImmutableMap.of(
+                        age, Domain.singleValue(INTEGER, 1L),
+                        score, Domain.onlyNull(DOUBLE)),
+                new BoolQueryBuilder()
+                        .filter(new TermQueryBuilder(age.getName(), 1))
+                        .mustNot(new ExistsQueryBuilder(score.getName())));
+    }
+
+    void assertQueryBuilder(Map<ElasticsearchColumnHandle, Domain> domains, QueryBuilder expected)
+    {
+        QueryBuilder res = buildSearchQuery(TupleDomain.withColumnDomains(domains), Optional.empty(), Map.of());
+        assertEquals(res.getClass(), expected.getClass());
+        assertEquals(res.toString(), expected.toString());
+        assertEquals(res.hashCode(), expected.hashCode());
```

Comment:
we can not use `assertEquals` directly, because `actual` and `expected` are same with value, not `object`. maybe we'd override `equals()` function of  `QueryBuilder` to compare with every elements.
here, simplify we use toString() and hashCode().

Reply:
Looks like equals are implemented in classes extending QueryBuilder. If you change the termquerybuilder to use 1L instead of 1 `testOneConstraint` at two places (isNullAllowed and SingleValue), all tests should pass comparing expected and actual.

Reply:
It looks better and I fix it. Before that, I didn't understand that `1` is wrong. 😊
thank you~